scripts/sct_deepseg.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+"""
+This command-line tool is the interface for the deepseg API that performs segmentation using deep learning from the
+ivadomed package.
+"""
+
+# TODO: Add link to example image so users can decide wether their images look "close enought" to some of the proposed
+#  models (e.g., mice, etc.).
+# TODO: add test to make sure that all postprocessing flags match the core.DEFAULT dictionary items
+# TODO: Fetch default value (and display) depending on the model that is used.
+
+from __future__ import absolute_import
+
+import sys
+import os
+import argparse
+import colored
+
+import spinalcordtoolbox as sct
+import spinalcordtoolbox.deepseg.core
+import spinalcordtoolbox.deepseg.models
+import spinalcordtoolbox.utils
+
+from sct_utils import init_sct, printv, display_viewer_syntax
+
+
+def get_parser():
+
+    parser = argparse.ArgumentParser(
+        description="Segmentation using deep learning.",
+        add_help=None,
+        formatter_class=sct.utils.SmartFormatter,
+        prog=os.path.basename(__file__).strip(".py"))
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        "-i",
+        help="Image to segment.",
+        metavar=sct.utils.Metavar.file)
+
+    seg = parser.add_argument_group('\nMODELS')
+    seg.add_argument(
+        "-m",
+        help="Model to use. For a description of each available model, please write: 'sct_deepseg -list-models'.",
+        choices=list(sct.deepseg.models.MODELS.keys()))
+    seg.add_argument(
+        "-list-models",
+        action='store_true',
+        help="Display a list of available models.")
+    seg.add_argument(
+        "-install-model",
+        help="Install specified model.",
+        choices=list(sct.deepseg.models.MODELS.keys()))
+    seg.add_argument(
+        "-install-default-models",
+        action='store_true',
+        help="Install all default models. Note: default models should already be installed during SCT installation.")
+    # TODO: replace URL below
+    seg.add_argument(
+        "-mpath",
+        help="Path to model, in case you would like to use a custom model. The model folder should follow the "
+             "conventions listed in: URL.",
+        metavar=sct.utils.Metavar.folder)
+
+    misc = parser.add_argument_group('\nPARAMETERS')
+    misc.add_argument(
+        "-thr",
+        type=float,
+        help="Binarize segmentation with specified threshold. Set to 0 for no thresholding (i.e., soft segmentation). "
+             "Default value is model-specific and was set during optimization "
+             "(more info at https://github.com/sct-pipeline/deepseg-threshold).",
+        metavar=sct.utils.Metavar.float)
+    misc.add_argument(
+        "-keep-largest-object",
+        type=int,
+        help="Remove false negative segmentation by only keeping the largest blob.",
```

Comment:
Oh! I misread this. I thought naively it would mean "include the largest object in the results" which, yes, obviously you would want that, but not that that meant "throw away all other objects". What about naming it just `-largest`?

I would also make this comment shorter, something like

```suggestion
        help="Keep only largest segment. This helps to reduce noise.",
```

Also I could see users wanting to tune this, so `-largest n` with `n` defaulting to 1?

Reply:
yup good idea-- partially implemented here 14a748dfad843e007c1182c56f5d4257c7c38a7f
i don,t have the time to implement the full feature